syntax = "proto3";

import "enums.proto";

package pslx;

message OperatorSnapShot {
    ModeType mode_type = 1;
    string operator_name = 2;
    DataModelType data_model = 3;
    Status status = 4;
}


message ContainerSnapShot {
    message GraphSnapShot {
        string node_name = 1;
        repeated string child_name = 2;
    }
    string root_name = 1;
    map<string, OperatorSnapShot> node_snapshot_map = 2;
    bool is_initialized = 3;
    Status status = 4;
}
